src/components/include/application_manager/rpc_service.h

```diff
@@ -54,10 +54,15 @@ class RPCService {
    * @brief ManageMobileCommand convert message to mobile command and execute it
    * @param message pointer to received message
    * @param origin origin of command
+   * @param warning_info warning message to send on a successful response. Only
+   * applies to requests from mobile.
    * @return true if command is executed, otherwise return false
    */
-  virtual bool ManageMobileCommand(const commands::MessageSharedPtr message,
-                                   commands::Command::CommandSource source) = 0;
+  virtual bool ManageMobileCommand(
```

Comment:
I do understand benefit of adding default argument, we will not have additional method and mock. 
But IMHO and according to [google coding stye](https://google.github.io/styleguide/cppguide.html#Default_Arguments) default arguments in virtual functions is bad practice. 
I would propose to make additional function `ManageMobileCommandExtended`/`ManageMobileCommandWarning`/`ManageMobileInfoCommand` ( naming is complicated) with warning code,  and keep ManageMobileCommand as proxy to it.

Reply:
Changed to overload in https://github.com/smartdevicelink/sdl_core/pull/3385/commits/446fe5c5c95865f6d7d088a088cbd5d655de2851